Source/FieldSolver/SpectralSolver/SpectralFieldDataRZ.cpp

```diff
@@ -509,15 +509,44 @@ SpectralFieldDataRZ::BackwardTransform (const int lev,
     // Loop over boxes.
     for (amrex::MFIter mfi(field_mf); mfi.isValid(); ++mfi){
 
-        amrex::Box const& realspace_bx = tempHTransformed[mfi].box();
+        amrex::Box realspace_bx = tempHTransformed[mfi].box();
 
         FABZBackwardTransform(mfi, realspace_bx, field_index, tempHTransformedSplit, is_nodal_z);
 
         // Perform the Hankel inverse transform last.
         // tempHTransformedSplit includes the imaginary component of mode 0.
         // field_mf does not.
         multi_spectral_hankel_transformer[mfi].SpectralToPhysical_Scalar(tempHTransformedSplit[mfi], field_mf_copy[mfi]);
-        field_mf[mfi].copy<amrex::RunOn::Device>(field_mf_copy[mfi], 0, i_comp*ncomp, ncomp);
+
+        amrex::Array4<amrex::Real> const & field_mf_array = field_mf[mfi].array();
+        amrex::Array4<amrex::Real> const & field_mf_copy_array = field_mf_copy[mfi].array();
+
+        // Extend the box to include the guards cells below the axis
+        // so that they can be filled in. This is not a simple copy
+        // since the signs will change when there is anti-symmetry.
+        amrex::Box const& realspace_bx_with_guards = field_mf[mfi].box();
+        const int* lo_with_guards = realspace_bx_with_guards.loVect();
+        realspace_bx.growLo(0, -lo_with_guards[0]);
```

Comment:
I think it would be really helpful if we could add a simple one-dimensional example with some numbers in the comments here. Just to show what kind of manipulation we are trying to do on a given box. What I have in mind is something along these lines (gaps must be filled in because I don't think I fully understand what's the goal here):
```cpp
// Example for a 1D box with 32 valid cells (starting from index 0) and 8 guard cells:
// 1) realspace_bx_with_guards = field_mf[mfi].box()
//    returns the box [-8,40] (that is, [-8,-1] U [0,32] U [33, 40])
// 2) realspace_bx_with_guards.loVect()
//    returns a pointer to the low end coordinate, in this case the integer ??
// 3) realspace_bx.growLo(0, -lo_with_guards[0])
//    returns the grown (or shrinked??) box [??,??] (in the direction 0)
```
Do you think it would be possible to add something like this and would you agree that it could be helpful?

Reply:
I updated the comments. Does this help?